pkg/controller/cronjob/utils.go

```diff
@@ -321,3 +321,16 @@ func adoptJobs(sj *batchv2alpha1.CronJob, js []batchv1.Job, jc jobControlInterfa
 	}
 	return utilerrors.NewAggregate(errs)
 }
+
+func getCurrentTimeInZone(sj *batchv2alpha1.CronJob) time.Time {
+	if sj.Spec.Timezone == nil {
+		return time.Now()
+	}
+
+	loc, err := time.LoadLocation(*sj.Spec.Timezone)
+	if err != nil {
+		loc = time.Local
```

Comment:
Yeah, validation sounds like a good idea. I think it's still possible that the list of time zones could change underneath the running controller process, in that case, we should probably not run the job at all and log an event.

As far as `time.Local`, previously we were just using `time.Now()` which returns the local time. So, in this case, the behavior is the same as previous. I would guess that most cluster operators would be using UTC, but, if they're not, they might expect that jobs are scheduled in the time zone of the controller manager.